src/common/translation.cpp

```diff
@@ -1887,12 +1887,28 @@ wxMsgCatalog *wxResourceTranslationsLoader::LoadCatalog(const wxString& domain,
     size_t mo_size = 0;
 
     const wxString resname = wxString::Format("%s_%s", domain, lang);
+    // Replace non-alphanumeric characters with `_`
+    wxString resname_safe = resname;
+    for ( wxString::iterator it = resname_safe.begin(); it != resname_safe.end(); ++it )
+    {
+        if ( !( (*it >= (wxChar) 'A' && *it <= (wxChar) 'Z') ||
+                (*it >= (wxChar) 'a' && *it <= (wxChar) 'z') ||
+                (*it >= (wxChar) '0' && *it <= (wxChar) '9') ||
+                *it == (wxChar) '_') )
```

Comment:
Can't we just use `wxIsalnum()` instead of all those comparisons? I also do not understand the check for underscore here.

Reply:
Yes, standard functions do depend on locale, so this is fine, but OTOH all these casts to `wxChar` are rather ugly. Why wouldn't something like
```cpp
const wxChar c = *it;
if ( !((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')) )
  *it = '_';
```
work instead?

And @PBfordev is right about the cast to `_` not being needed here, of course.